L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc

```diff
@@ -309,25 +309,10 @@ void L1TkFastVertexProducer::produce(edm::StreamID, edm::Event& iEvent, const ed
 
   // sliding windows... maximize bin i + i-1  + i+1
 
-  float zvtx_sliding = -999;
-  float sigma_max = -999;
-  int imax = -999;
+  float zvtx_sliding;
+  float sigma_max;
+  int imax;
   int nb = htmp.GetNbinsX();
-  for (int i = 2; i <= nb - 1; i++) {
```

Comment:
All this loop compute uselessly variables which are recompute anyhow in the following one.
The trivial fix wold be to remove this loop as a whole, as suggested here: but perhaps the author of the code hade something else in mind, so better to have an expert look.

Reply:
You are correct that this loop is unnecessary, and removing it would be fine. However, this is the old vertex producer that we're migrating away from, and will soon be obsolete. The newer producer we are using going forward is L1Trigger/VertexFinder

Reply:
Let's remove the loop now, then, since I'm not entirely sure when this will become strictly obsolete.